libs/vkd3d/command.c

```diff
@@ -9745,12 +9732,12 @@ static void STDMETHODCALLTYPE d3d12_command_list_RSSetShadingRateImage(d3d12_com
     if (vrs_image == list->vrs_image)
         return;
 
-    d3d12_command_list_invalidate_current_framebuffer(list);
+    d3d12_command_list_invalidate_rendering_info(list);
 
     /* Need to end the renderpass if we have one to make
      * way for the new VRS attachment */
-    if (list->current_render_pass || list->render_pass_suspended)
-        d3d12_command_list_invalidate_current_render_pass(list);
+    if (list->rendering_info.state_flags & (VKD3D_RENDERING_ACTIVE | VKD3D_RENDERING_SUSPENDED))
+        d3d12_command_list_end_current_render_pass(list, false);
```

Comment:
Below, the pipeline is invalidated on `if (!list->vrs_image != !vrs_image)`. This is now obsolete.

Reply:
fixed.